### PR TITLE
[feat]:Adding Tags and Priority to the Note

### PR DESCRIPTION
**Pull request by ultronjs**
-Adding Tags and Priority to the new node
-Editing the tags and Priority in existing note.

### CODE REVIEW

**Review by bajajcodes — COMMENTED** [review]
- Good amount of features, inspired from you.
- Code style is consistent.
- Good understanding of git, as files changed actual, not bundle.

## Suggestions
- Add annotation on commented code, describing why its there in place.
- Remove console logs.

File: src/components/AddNote.jsx [hunk 1]
```diff
@@ -5,22 +5,37 @@ import { BsPin, BsFillPinFill } from "react-icons/bs";
 import { MdOutlineColorLens, MdColorLens } from "react-icons/md";
 import { BiArchiveIn, BiArchiveOut } from "react-icons/bi";
 import {GrPowerReset} from "react-icons/gr"
-import {MdLabelOutline} from "react-icons/md"
+import { MdLabelOutline, MdLabel } from "react-icons/md";
```
Comment (t007rushi) [inline src/components/AddNote.jsx]: can make icons.js for single imports

File: src/components/AddNote.jsx [hunk 2]
```diff
@@ -5,22 +5,37 @@ import { BsPin, BsFillPinFill } from "react-icons/bs";
...
+import Select from "react-select";
 function AddNote() {
+  const tagOptions = [
+    { value: "Home", label: "Home" },
+    { value: "Work", label: "Work"},
+    { value: "Hobby", label: "Hobby"},
+    { value: "Study", label: "Study"},
+    { value: "Passion", label: "Passion"  },
+    { value: "Preparation", label: "Preparation" },
+    { value: "Others", label: "Others"  }
```
Comment (t007rushi) [inline src/components/AddNote.jsx:24]: can work on dynamic tags/labels from user inputs

File: src/components/AddNote.jsx [hunk 3]
```diff
@@ -5,22 +5,37 @@ import { BsPin, BsFillPinFill } from "react-icons/bs";
...
+  ];
+  const priorityOptions = [
+    { value: "Low", label: "Low", className: "badge_success" },
+    { value: "Medium", label: "Medium", className: "badge_warning" },
+    { value: "High", label: "High", className: "badge_danger" },
```
Comment (t007rushi) [inline src/components/AddNote.jsx:29]: same here

File: src/components/AddNote.jsx [hunk 4]
```diff
@@ -5,22 +5,37 @@ import { BsPin, BsFillPinFill } from "react-icons/bs";
...
   const noteInitialObj = {
     title: "",
     description: "",
     pinned: false,
     color: "",
-    label: "",
-    archive:false,
-    tags:[],
-    createdAt:new Date(),
+    archive: false,
```
Comment (t007rushi) [inline src/components/AddNote.jsx:36]: there is no need of this as archive and notes has diff routes

File: src/components/AddNote.jsx [hunk 5]
```diff
@@ -30,9 +45,16 @@ function AddNote() {
...
+  const changeLabel = (e) => {
+    console.log(e)
+    setNote((prevState) => ({
+      ...prevState,
+      tags: e,
+    }));
+  }
   console.log(note)
   // const{children}=parse(content).props
   // console.log(children)
```
Comment (t007rushi) [inline src/components/AddNote.jsx]: remove comments

File: src/components/AddNote.jsx [hunk 6]
```diff
@@ -30,9 +45,16 @@ function AddNote() {
...
+    setNote((prevState) => ({ ...prevState, priority: e }));
   };
+  const changeLabel = (e) => {
+    console.log(e)
+    setNote((prevState) => ({
+      ...prevState,
+      tags: e,
+    }));
+  }
   console.log(note)
```
Comment (t007rushi) [inline src/components/AddNote.jsx]: remove consoles's

File: src/components/AddNote.jsx [hunk 7]
```diff
@@ -52,38 +74,81 @@ function AddNote() {
...
+                  <option className="selection_menu_option" value="Orange">
+                    Orange
+                  </option>
+                  <option className="selection_menu_option" value="Radish">
+                    Radish
+                  </option>
+                  <option className="selection_menu_option" value="Cherry">
+                    Cherry
+                  </option>
+                </select> */}
```
Comment (t007rushi) [inline src/components/AddNote.jsx:133]: same here

File: src/components/AddNote.jsx [hunk 8]
```diff
@@ -52,38 +74,81 @@ function AddNote() {
           <BsPin onClick={changePinnedStatus} size={25} />
         )}
         <Editor content={content} setContent={setContent} />
-        <div className="add_note_footer flex flex-jc-flex-end flex-ai-center gap-m mr-small">
-          {showColorPallete ? (
-            <MdColorLens
-              size={25}
-              onClick={() => setColorPallete((prevStatus) => !prevStatus)}
+        <div className="add_note_footer flex flex-jc-space-between">
```
Comment (t007rushi) [inline src/components/AddNote.jsx:77]: some time ```_``` underscore & some time ```-``` dash keep a consistent naming in className

File: src/components/EditNote.jsx [hunk 9]
```diff
@@ -4,26 +4,43 @@ import "../index.css";
...
+    { value: "Passion", label: "Passion" },
+    { value: "Preparation", label: "Preparation" },
+    { value: "Others", label: "Others" },
+  ];
+  const priorityOptions = [
+    { value: "Low", label: "Low", className: "badge_success" },
+    { value: "Medium", label: "Medium", className: "badge_warning" },
+    { value: "High", label: "High", className: "badge_danger" },
+  ];
   console.log(noteDetails)
```
Comment (t007rushi) [inline src/components/EditNote.jsx]: or you can put them aside as its static code also console here

**Review by t007rushi — APPROVED** [review]
Look for comments and console before push

File: src/components/AddNote.jsx (hunk 4 above)
Comment (ultronjs) [reply src/components/AddNote.jsx:36]: It is required as i am using as same note component to display active note and archive note.Archive icon is based on this flag

File: src/components/AddNote.jsx (hunk 3 above)
Comment (ultronjs) [reply src/components/AddNote.jsx:29]: Making priority also dynamic??

File: src/components/AddNote.jsx (hunk 2 above)
Comment (ultronjs) [reply src/components/AddNote.jsx:24]: WIll do this in next PR

File: src/components/EditNote.jsx (hunk 9 above)
Comment (ultronjs) [reply src/components/EditNote.jsx]: Will move this in utils

File: src/components/AddNote.jsx (hunk 1 above)
Comment (ultronjs) [reply src/components/AddNote.jsx]: Thanks for the suggestion

File: src/components/AddNote.jsx (hunk 8 above)
Comment (ultronjs) [reply src/components/AddNote.jsx:77]: _(underscore) is for the class which are in this project whereas -(dash) is for the class which are coming from Component Library

File: src/components/AddNote.jsx (hunk 7 above)
Comment (ultronjs) [reply src/components/AddNote.jsx:133]: Done.Resolved

File: src/components/AddNote.jsx (hunk 6 above)
Comment (ultronjs) [reply src/components/AddNote.jsx]: Done.Resolved

File: src/components/AddNote.jsx (hunk 5 above)
Comment (ultronjs) [reply src/components/AddNote.jsx]: Done

File: src/components/AddNote.jsx (hunk 1 above)
Comment (ultronjs) [reply src/components/AddNote.jsx]: Done

**Comment by ultronjs** [comment]
Merging the request after 2 approval and resolving git comments

**Pull request merged**